### PR TITLE
Helper type for CSS parser

### PR DESCRIPTION
**Pull request by Jym77**
Part of #1202

### CODE REVIEW

**Comment by Jym77** [comment]
!pr extract

**Comment by Jym77** [comment]
!pr extract

File: packages/alfa-css/src/calculation/numeric/angle.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/calculation/numeric/angle.ts:3]: **suggestion:** It looks like `Slice` is no longer used
```suggestion

```

File: packages/alfa-css/src/calculation/numeric/integer.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/calculation/numeric/integer.ts:3]:
```suggestion

```

File: packages/alfa-css/src/calculation/numeric/length.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/calculation/numeric/length.ts:3]:
```suggestion

```

File: packages/alfa-css/src/calculation/numeric/number.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/calculation/numeric/number.ts:2]:
```suggestion

```

File: packages/alfa-css/src/calculation/numeric/percentage.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/calculation/numeric/percentage.ts:2]:
```suggestion

```

File: packages/alfa-css/src/syntax/nth.ts
Comment (rcj-siteimprove) [inline packages/alfa-css/src/syntax/nth.ts:1]: **suggestion:** `import { Slice } ...` can be removed

File: packages/alfa-css/src/value/collection/list.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { Mapper } from "@siteimprove/alfa-mapper";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/collection/list.ts:6]:
```suggestion

```

File: packages/alfa-css/src/value/image/gradient.ts [hunk 7]
```diff
@@ -207,15 +206,10 @@ export namespace Gradient {
     export type JSON = Stop.JSON | Hint.JSON;
   }
 
-  export const parseItem: Parser<Slice<Token>, Item, string> = either(
-    parseStop,
-    parseHint
-  );
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/image/gradient.ts:213]: **question:** Is this a breaking change that requires a change set?

File: packages/alfa-css/src/value/shape/ellipse.ts [hunk 8]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/shape/ellipse.ts:3]:
```suggestion

```

File: packages/alfa-css/src/value/shape/inset.ts [hunk 9]
```diff
@@ -5,7 +5,7 @@ import { Option } from "@siteimprove/alfa-option";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/shape/inset.ts:6]:
```suggestion

```

File: packages/alfa-css/src/value/shape/rectangle.ts [hunk 10]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/shape/rectangle.ts:3]:
```suggestion

```

File: packages/alfa-css/src/value/transform/translate.ts [hunk 11]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/transform/translate.ts:3]:
```suggestion

```

File: packages/alfa-css/src/value/keyword.ts [hunk 12]
```diff
@@ -3,7 +3,7 @@ import { Parser } from "@siteimprove/alfa-parser";
 import { Predicate } from "@siteimprove/alfa-predicate";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/keyword.ts:4]:
```suggestion

```

File: packages/alfa-css/src/value/position.ts
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/position.ts:1]: **suggestion:** `import { Slice } ...` can be removed

File: packages/alfa-css/src/value/shadow.ts
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/shadow.ts:1]: **suggestion:** `import { Parser } ...` and `import { Slice } ...` can be removed

File: packages/alfa-css/src/value/string.ts [hunk 13]
```diff
@@ -2,7 +2,7 @@ import { Hash } from "@siteimprove/alfa-hash";
 import { Parser } from "@siteimprove/alfa-parser";
 import { Slice } from "@siteimprove/alfa-slice";
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/string.ts:3]:
```suggestion

```

**Review by rcj-siteimprove — COMMENTED** [review]
It looks like the refactor has made some import statements redundant

**Review by rcj-siteimprove — APPROVED** [review]

**Pull request merged**